environment/src/main/java/com/sequenceiq/environment/credential/service/CredentialPrerequisiteService.java

```diff
@@ -131,13 +135,23 @@ public Credential decorateCredential(Credential credential) {
         CredentialAttributes credentialAttributes = getCredentialAttributes(credential);
         if (isRoleArnSet(credentialAttributes)) {
             if (credentialAttributes.getAws().getRoleBased().getExternalId() == null) {
-                credentialAttributes.getAws().getRoleBased().setExternalId(userPreferencesService.getExternalIdForCurrentUser());
+                credentialAttributes.getAws().getRoleBased().setExternalId(getExternalId(credential.getType()));
                 saveNewAttributesToCredential(credential, credentialAttributes);
             }
         }
         return credential;
     }
 
+    private String getExternalId(CredentialType type) {
+        String externalId = null;
+        if (CredentialType.AUDIT.equals(type)) {
+            externalId = userPreferencesService.getExternalIdForCurrentUser();
```

Comment:
I think `userPreferencesService.getAuditExternalIdForCurrentUser();` should be called here by credential type, so can we swap the 148 and 150 line?